src/MetaMorpho.sol

```diff
@@ -202,6 +202,10 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
         if (newTimelock > timelock) {
             _setTimelock(newTimelock);
         } else {
+            if (pendingTimelock.submittedAt != 0 && newTimelock == pendingTimelock.value) {
```

Comment:
Is the first check necessary?

How can we have `pendingTimelock.value != 0` but `pendingTimelock.submittedAt == 0`

Reply:
I believe we are not looking to study `pendingTimelock.submittedAt == 0 && pendingTimelock.value != 0` but instead `pendingTimelock.submittedAt == 0 && pendingTimelock.value == newTimelock`

We know invariant `pendingTimelock.submittedAt == 0 && pendingTimelock.value == 0`

For some inputs, `newTimelock` may be `0` which would satisfy `pendingTimelock.submittedAt == 0 && pendingTimelock.value == newTimelock == 0`

I'll think for each of the 4 timelocked values:
- timelock: `0 < MIN_TIMELOCK <= newTimelock <= MAX_TIMELOCK` so impossible
- fee: `newFee` may equal `0`
- guardian: `newGuardian` may equal `address(0)`
- cap: `newCap` may equal `0`

Reply:
Indeed for the `fee`, `guardian` and `cap`

Reply:
well in fact I'm not sure, if `newFee` is 0, then it's not pending no? Same for the cap, right?